client/executor/common/src/wasmer_backend.rs

```diff
@@ -0,0 +1,433 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Wasmer specific impls for sandbox
+
+use crate::{
+	error::Result,
+	sandbox::Memory,
+	util::{checked_range, MemoryTransfer},
+};
+use codec::Encode;
+use sp_wasm_interface::{FunctionContext, Pointer, Value, WordSize};
+use std::{cell::RefCell, collections::HashMap, convert::TryInto, rc::Rc};
+use wasmi::RuntimeValue;
+
+use crate::{
+	error::Error,
+	sandbox::{
+		deserialize_result, BackendInstance, GuestEnvironment, InstantiationError, SandboxContext,
+		SandboxInstance, SupervisorFuncIndex,
+	},
+};
+
+environmental::environmental!(SandboxContextStore: trait SandboxContext);
+
+/// Wasmer specific context
+pub struct WasmerBackend {
```

Comment:
Should just called be `Backend` as it is already in the wasmer module. The same for all similar type in both backends.